aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -41,6 +41,9 @@ static inline Tensor integer_upcast(const Tensor& self, optional<ScalarType> dty
 using DimMask = TensorIterator::DimMask;
 
 static DimMask make_dim_mask(IntArrayRef dims, int64_t ndim) {
+  for(size_t idx =0; idx=dims.size(); idx++) {
```

Comment:
Also, instead of doing the iteration here, could we check if `dim < 64` on line51 instead? https://github.com/pytorch/pytorch/pull/31476/files#diff-847680695fb0535da86db8b98a13b5a3R51-R53

Reply:
good idea! updated the code